base/range.jl

```diff
@@ -801,3 +801,5 @@ end
 
 in{T<:Integer}(x, r::Range{T}) = isinteger(x) && !isempty(r) && x>=minimum(r) && x<=maximum(r) && (mod(convert(T,x),step(r))-mod(first(r),step(r)) == 0)
 in(x::Char, r::Range{Char}) = !isempty(r) && x >= minimum(r) && x <= maximum(r) && (mod(Int(x) - Int(first(r)), step(r)) == 0)
+
+unique{T}(x::Range{T}) = step(x) == zero(T) ? x[1:1] : x
```

Comment:
Interesting. My initial reaction is that this is actually an issue with `LinSpace` and/or the semantics `step` method, but I wasn't able to find documentation on the intended behavior for `step`. One alternative implementation strategy which would capture the case you bring up here would be to write like
```julia
_unique{T<:AbstractFloat}(x::Range{T}) = x[1] == x[end] ? x : construct appropriate range
```
Or do you think there are additional cases where there are more than two unique elements for float-like ranges?